MetaMask/metamask-extension
# feat(srp): add a quiz to the SRP reveal

## Explanation

* Fixes #17557
* Fixes https://github.com/MetaMask/MetaMask-planning/issues/597
* Fixes https://github.com/MetaMask/MetaMask-planning/issues/268

[Figma file](https://www.figma.com/file/4LQje3epQz8g4xO8fRdauv/SRP-Reveal-Warning)
[Google Doc](https://docs.google.com/document/d/1kzI0BiBT9-6LuEp19lfZu0RV7wnXLOX24zg0r4uxK0A)

Uses the DS component `ui/components/component-library/modal`

**100% test coverage for added files and for `/ui/pages/settings/security-tab` folder**

## Screenshots/Screencaps
### After
![image](https://github.com/MetaMask/metamask-extension/assets/539738/61854282-2b5c-4864-bc46-1c76d07c826e)

## Manual Testing Steps

- Go to Settings -> Security & privacy
- Click Reveal Secret Recovery Phrase
- Get the questions right, get the questions wrong, make sure it goes to the right place
- If you have access to MixPanel, make sure it fires the right events at the right time

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
